packages/zoe/test/unitTests/contracts/newSwap/test-newswap-bug.js

```diff
@@ -0,0 +1,221 @@
+    'RUN',
+    AssetKind.NAT,
+    harden({ decimalPlaces: 6 }),
+  );
+  const bldKit = makeIssuerKit(
+    'BLD',
+    AssetKind.NAT,
+    harden({ decimalPlaces: 6 }),
+  );
+  const zoe = makeZoe(fakeVatAdmin);
+
+  // Pack the contract.
+  const bundle = await bundleSource(multipoolAutoswapRoot);
+
+  const installation = await zoe.install(bundle);
+  const fakeTimer = buildManualTimer(console.log, 30n);
+  const { publicFacet } = await zoe.startInstance(
+    installation,
+    harden({ Central: runKit.issuer }),
+    {
+      timer: fakeTimer,
+      poolFee: DEFAULT_POOL_FEE,
+      protocolFee: DEFAULT_PROTOCOL_FEE,
+    },
+  );
+  const aliceAddLiquidityInvitation = E(
+    publicFacet,
+  ).makeAddLiquidityInvitation();
+
+  const bldLiquidityIssuer = await E(publicFacet).addPool(bldKit.issuer, 'BLD');
+  const bldLiquidityBrand = await E(bldLiquidityIssuer).getBrand();
+
+  const bldPoolAllocation = AmountMath.make(bldKit.brand, 2196247730468n);
+  const runPoolAllocation = AmountMath.make(runKit.brand, 50825056949339n);
+
+  const aliceProposal = harden({
+    give: {
+      Secondary: bldPoolAllocation,
+      Central: runPoolAllocation,
+    },
+    want: { Liquidity: AmountMath.make(bldLiquidityBrand, 0n) },
+  });
+
+  const alicePayments = {
+    Secondary: bldKit.mint.mintPayment(bldPoolAllocation),
+    Central: runKit.mint.mintPayment(runPoolAllocation),
+  };
+
+  const addLiquiditySeat = await zoe.offer(
+    aliceAddLiquidityInvitation,
+    aliceProposal,
+    alicePayments,
+  );
+
+  t.is(
+    await E(addLiquiditySeat).getOfferResult(),
+    'Added liquidity.',
+    `Alice added bld and run liquidity`,
+  );
+
+  await addLiquiditySeat.getPayout('Liquidity');
+
+  const priceQuote = await E(publicFacet).getPriceGivenAvailableInput(
+    AmountMath.make(runKit.brand, 73000000n),
+    bldKit.brand,
+  );
+
+  console.log('price quote', priceQuote);
+
+  const percentToReduceBy = makeRatio(
+    reduceWantOutBP,
+    priceQuote.amountOut.brand,
+    BASIS_POINTS,
+  );
+  const wantOutReduced = AmountMath.subtract(
+    priceQuote.amountOut,
+    multiplyBy(priceQuote.amountOut, percentToReduceBy),
+  );
+
+  const proposal = harden({
+    give: { In: priceQuote.amountIn },
+    want: { Out: wantOutReduced },
+  });
+
+  const payment = harden({ In: runKit.mint.mintPayment(priceQuote.amountIn) });
+
+  const seatP = E(zoe).offer(
+    E(publicFacet).makeSwapInInvitation(),
+    proposal,
+    payment,
+  );
+
+  const offerResult = await E(seatP).getOfferResult();
+
+  console.log(offerResult);
+};
+
+test('test bug scenario with 0% reduction in want', async t => {
+  await conductTrade(t, 0n);
+});
+
+test('test bug scenario with 0.2% reduction in want', async t => {
+  await conductTrade(t, 20n);
+});
+
+test('test bug scenario with 0.3% reduction in want', async t => {
+  await conductTrade(t, 30n);
+});
+
+test('test bug scenario with 3% reduction in want', async t => {
+  await conductTrade(t, 300n);
+});
```

Comment:
I don't think we need to include these tests. They were only for trying to figure out whether the bug could be fixed by a UI change alone. Now that we don't need to figure that out, we can drop them.

Reply:
done